src/test/ui/resolve/issue-22692.stderr

```diff
@@ -0,0 +1,9 @@
+error[E0423]: expected value, found struct `String`
+  --> $DIR/issue-22692.rs:12:13
+   |
+12 |     let _ = String.new(); //~ ERROR did you mean to call static method `new`?: `String::new`
+   |             ^^^^^^---- help: did you mean to call static method `new`?: `String::new`
```

Comment:
great. but it'll be better to have a suggestion here with https://doc.rust-lang.org/nightly/nightly-rustc/syntax/diagnostics/plugin/struct.Diagnostic.html#method.span_suggestion_with_applicability

Reply:
Specifically, the last argument to `span_suggestion_with_applicability` should be `Applicability::MaybeIncorrect`. (See the [diagnostics chapter of the Contributor's Guide](https://rust-lang-nursery.github.io/rustc-guide/diag.html#suggestions) and #50723 for context.)

Reply:
Please make sure to cover the correct range even when this is "MaybeIncorrect", people can in theory still apply this manually with RLS. (I.e., cover exactly `String.new` if you suggest `String::new`, or `String.new()` if you suggest `String::new()`)

Reply:
Note: I'm not sure, I would have to look at the replaced generated file, but it seems to me that the correct span is covered (the `-` underline is just covered by the `^`, as intemded).